# Challenge - Beware of Monty Hall

You are on a game show and given the choice of whatever is behind three doors. Behind one door is a fantastic prize (some examples use a car, others use cash) while behind the other two doors is a dud (some examples say a goat, others say it's just empty). You pick a door. Then the host opens one of the other two doors to reveal a dud. But here's the wrinkle: the host now gives you the opportunity to switch your door. What should you do?

In [6]:
# Before host opens door
prob_one = 1/3
prob_two = 1/3
prob_three = 1/3

# After host opens door
new_prob_three = 0
new_prob_one = 1/3
new_prob_two = 1 - (new_prob_one + new_prob_three)

print(new_prob_two)

0.6666666666666667
True


Because the probability of the 2nd door holding the car (0.667) is larger than the probability of your original choice holding the car (0.333), it is advantageous to switch your door.